Local installation broken on Windows

I tried to perform the [local installation](https://github.com/OpenLightingProject/open-fixture-library/blob/master/docs/CONTRIBUTING.md#local-installation) on Windows today. I wasn't able to get it working and ultimately had to change to an Ubuntu VM. 

A list of the errors I encountered:

## No Make
During `npm install` in Git Bash:
```
> open-fixture-library@1.0.0 postinstall C:\Users\Johannes\Documents\Open Fixture Library dev\open-fixture-library
> make all

Der Befehl "make" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! open-fixture-library@1.0.0 postinstall: `make all`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the open-fixture-library@1.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```
Windows does not come with make. I installed make via Chocolatey and the error went away. 

## Spaces in File Path
During `npm install` in Git Bash:
```
$(npm bin)/nuxt build
/usr/bin/sh: C:\Users\Johannes\Documents\Open: No such file or directory
make: *** [Makefile:30: nuxt-build] Error 127
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! open-fixture-library@1.0.0 postinstall: `make all`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the open-fixture-library@1.0.0 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```
Repository folder was `C:\Users\Johannes\Documents\Open Fixture Library dev\open-fixture-library`. It contained spaces and that clearly broke nuxt-build. 
Renaming project folder without spaces resolved the issue. 

## Wrong loaders

<details>
<summary>Fixed in <a href="https://github.com/OpenLightingProject/open-fixture-library/pull/1278">#1278</a>.</summary>

<blockquote>
<p>After fixing the errors above I could get the server to start, but all the SVG icons on the home page were replaced by jumbled text. For the example the folder icon next to "Browse Fixtures" was rendered as</p>

<pre>data:image/svg+xml;base64,bW9kdWxlLmV4cG9ydHMgPSAiPHN2ZyB4bWxucz1cImh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnXCIgdmVyc2lvbj1cIjEuMVwiIHZpZXdCb3g9XCIwIDAgMjQgMjRcIj48cGF0aCBkPVwiTTIyLDRIMTRMMTIsMkg2QTIsMiAwIDAsMCA0LDRWMTZBMiwyIDAgMCwwIDYsMThIMjJBMiwyIDAgMCwwIDI0LDE2VjZBMiwyIDAgMCwwIDIyLDRNMiw2SDBWMTFIMFYyMEEyLDIgMCAwLDAgMiwyMkgyMFYyMEgyVjZaXCI+PC9wYXRoPjwvc3ZnPiI=</pre>

<p>This seems to be a data URI rendered as text inside an HTML element. 
In contrast, the online homepage shows:</p>

<pre>&lt;svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="24" height="24" viewBox="0 0 24 24"&gt;&lt;path d="M22,4H14L12,2H6A2,2 0 0,0 4,4V16A2,2 0 0,0 6,18H22A2,2 0 0,0 24,16V6A2,2 0 0,0 22,4M2,6H0V11H0V20A2,2 0 0,0 2,22H20V20H2V6Z"&gt;&lt;/path&gt;&lt;/svg&gt;</pre>

<p>which gets properly displayed as image. 
I tracked the problem to some wrong loader behaviour in webpack/Nuxt. The data URI is probably the result of some combination of <code>svg-inline-loader</code> and <code>url-loader</code>. Commenting out the svg-exclude for <code>url-loader</code> in 
<a href="https://github.com/OpenLightingProject/open-fixture-library/blob/b4ddc16dce660982e735d705b03f34689406a038/nuxt.config.js#L37-L39">nuxt.config.js#L37-L39</a>
changes nothing.  
Also commenting out 
<a href="https://github.com/OpenLightingProject/open-fixture-library/blob/b4ddc16dce660982e735d705b03f34689406a038/nuxt.config.js#L41-L51">nuxt.config.js#L41-L51</a>
results in only <code>url-loader</code> being used, which gives a different data URI output still than when <code>svg-inline-loader</code> is activated:</p>

<pre>data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIgd2lkdGg9IjI0IiBoZWlnaHQ9IjI0IiB2aWV3Qm94PSIwIDAgMjQgMjQiPjxwYXRoIGQ9Ik0yMiw0SDE0TDEyLDJINkEyLDIgMCAwLDAgNCw0VjE2QTIsMiAwIDAsMCA2LDE4SDIyQTIsMiAwIDAsMCAyNCwxNlY2QTIsMiAwIDAsMCAyMiw0TTIsNkgwVjExSDBWMjBBMiwyIDAgMCwwIDIsMjJIMjBWMjBIMlY2WiIgLz48L3N2Zz4NCg</pre>

<p>I couldn't get Nuxt/webpack to only use <code>svg-inline-loader</code> without resorting to an inline overwrite like <code>!!svg-inline-loader!./file-to-import.svg</code>.</p>
</blockquote>
</details>

## No Windows binary for libxmljs
I tried the local install again, but this time instead of Git Bash I used Windows cmd. During `npm install` I got: 
```
E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library>npm install

> libxmljs@0.19.7 install E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs
> node-pre-gyp install --fallback-to-build --loglevel http

node-pre-gyp WARN Using request for node-pre-gyp https download
node-pre-gyp http GET https://github.com/libxmljs/libxmljs/releases/download/v0.19.7/node-v72-win32-x64.tar.gz
node-pre-gyp http 404 https://github.com/libxmljs/libxmljs/releases/download/v0.19.7/node-v72-win32-x64.tar.gz
node-pre-gyp WARN Tried to download(404): https://github.com/libxmljs/libxmljs/releases/download/v0.19.7/node-v72-win32-x64.tar.gz
node-pre-gyp WARN Pre-built binaries not found for libxmljs@0.19.7 and node@12.16.2 (node-v72 ABI, unknown) (falling back to source compile with node-gyp)
node-pre-gyp http 404 status code downloading tarball https://github.com/libxmljs/libxmljs/releases/download/v0.19.7/node-v72-win32-x64.tar.gz
Traceback (most recent call last):
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\gyp_main.py", line 50, in <module>
    sys.exit(gyp.script_main())
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 554, in script_main
    return main(sys.argv[1:])
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 547, in main
    return gyp_main(args)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 532, in gyp_main
    generator.GenerateOutput(flat_list, targets, data, params)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 2008, in GenerateOutput
    msvs_version)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 1843, in _CreateProjectObjects
    guid = _GetGuidOfProject(proj_path, spec)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\generator\msvs.py", line 910, in _GetGuidOfProject
    guid = guid or MSVSNew.MakeGuid(proj_path)
  File "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\MSVSNew.py", line 48, in MakeGuid
    d = hashlib.md5((str(seed) + str(name)).encode('utf-8')).hexdigest().upper()
UnicodeDecodeError: 'ascii' codec can't decode byte 0xfc in position 22: ordinal not in range(128)
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--loglevel=http" "--module=E:\\ZickeZackeHühnerkacke\\Open Fixture Library\\open-fixture-library\\node_modules\\libxmljs\\build\\Release\\xmljs.node" "--module_name=xmljs" "--module_path=E:\\ZickeZackeHühnerkacke\\Open Fixture Library\\open-fixture-library\\node_modules\\libxmljs\\build\\Release" "--napi_version=5" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v72" "--python=C:\\Users\\Johannes\\.windows-build-tools\\python27\\python.exe"
gyp ERR! cwd E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs
gyp ERR! node -v v12.16.2
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --loglevel=http --module=E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs\build\Release\xmljs.node --module_name=xmljs --module_path=E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs\build\Release --napi_version=5 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72 --python=C:\Users\Johannes\.windows-build-tools\python27\python.exe' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:310:20)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:286:5)
node-pre-gyp ERR! System Windows_NT 10.0.18362
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "E:\\ZickeZackeHühnerkacke\\Open Fixture Library\\open-fixture-library\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build" "--loglevel" "http"
node-pre-gyp ERR! cwd E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs
node-pre-gyp ERR! node -v v12.16.2
node-pre-gyp ERR! node-pre-gyp -v v0.11.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --loglevel=http --module=E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs\build\Release\xmljs.node --module_name=xmljs --module_path=E:\ZickeZackeHühnerkacke\Open Fixture Library\open-fixture-library\node_modules\libxmljs\build\Release --napi_version=5 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v72 --python=C:\Users\Johannes\.windows-build-tools\python27\python.exe' (1)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules\nunjucks\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.12 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.12: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! libxmljs@0.19.7 install: `node-pre-gyp install --fallback-to-build --loglevel http`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the libxmljs@0.19.7 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```
There really seems to be no current binary release of libxmljs available for windows and node-v72 (https://github.com/libxmljs/libxmljs/releases). 

### Environment
```
npm -v
6.14.4

node -v
v12.16.2
```
In all cases, the typical solutions of deleting the folder node_modules or clearing the npm cache changed nothing. 

In the end I set up a new Ubuntu 20.04 VM, in which both install and the the icons seemed to function perfectly. 

I know this is a cluttered and undetailed report, but I'm new to the project and the technology stack and uncertain how to properly handle these issues or to what extent they belong together. 
However, installing under Windows was a bad and buggy experience for me. If we can't fix these issues, maybe we could put a note in `CONTRIBUTING.md` that Windows is not (yet?) a supported platform?

--- a/docs/CONTRIBUTING.md
+++ b/docs/CONTRIBUTING.md
@@ -14,6 +14,8 @@ As soon as an issue is assigned to somebody, it means that this person is respon
 
 ## Local installation
 
+*Note for Windows users:* Local installation is currently broken on Windows. See [#1276](https://github.com/OpenLightingProject/open-fixture-library/issues/1276).
+
 1. Install [npm](https://www.npmjs.com/get-npm)
 2. [Clone](https://help.github.com/articles/cloning-a-repository/) the repository
 3. Navigate into your clone's directory (probably `open-fixture-library/`)

